diff -aru netperf-2.4.5/src/netlib.h netperf-2.4.5/src/netlib.h
--- netperf-2.4.5/src/netlib.h	2008-10-30 16:48:46.000000000 -0400
+++ netperf-2.4.5/src/netlib.h	2012-04-10 20:30:48.202614923 -0400
@@ -16,7 +16,7 @@
 # include "missing/getaddrinfo.h"
 #endif
 
-#define PAD_TIME 4
+#define PAD_TIME 30
 /* library routine specifc defines                                      */
 #define         MAXSPECDATA     62      /* how many ints worth of data  */
                                         /* can tests send...            */
